### ISSUE TITLE
Inserter should be fully keyboardable

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#87]
Inspired by solid work in https://github.com/WordPress/gutenberg/pull/85, let's take the inserter dialog a bit further.

![screen shot 2017-02-17 at 12 34 16](https://cloud.githubusercontent.com/assets/1204802/23064082/4e501aec-f50e-11e6-8c17-1b507d000190.png)

The model to emulate is an amalgam of Slack (when typing / in the message box) and the OSX Emoji inserter (invoke by pressing ^⌘ Space). 

As soon as you invoke the inserter, you can use the arrow keys to pick a block and press enter to insert. If you start typing, you start filtering. Type `ima` and you'll probably narrow it down to an Image being the only block left. It has the focus rectangle, so when you press Enter, it's inserted.

To put it simply, if you're a keyboard shortcut person, you should be able to:

1. Type `/` on a new line to insert a block (see #24)
2. Type `ima` to filter down to just the Image
3. Press Enter to insert the image block

**Comment by youknowriad** [comment]
I have a question regarding this @jasmussen 

Do you think we can have the same block inside to different sections. Say Image can leave in "Frequently Used" and also in "Media" which could have impact on the implementation details here. Do we need to store the selected block only or a tuple (selected section, selected block)?

**Comment by jasmussen** [comment]
Excellent question. The recent label suggests indeed that it exists twice, or that those in recent are "symlinked" references.

However @folletto reminded me recently that when the section is rearranged based on usage, you can't develop muscle memory or spatial awareness for where the blocks you use the most sit. 

So perhaps for simplicity, and in the name of shipping, let's for now rename the section to "common", and have it be a curated list.

Does that answer your question, or give you what you need to keep moving?

**Comment by youknowriad** [comment]
Yep, this sounds good for me. Tks a lot 👍
